guacamole-common-js/src/main/webapp/modules/Tunnel.js

```diff
@@ -215,6 +215,24 @@ Guacamole.HTTPTunnel = function(tunnelURL, crossDomain) {
     var receive_timeout = null;
 
     /**
+     * Additional headers to be sent in tunnel requests. This dictionary can be
+     * populated with key/value header pairs to pass information such as authentication
+     * tokens, etc.
+     */
+    var extraHeaders = {};
+
+    /**
+     * Adds the configured additional headerss to the given request.
+     *
+     * @private
+     */
+    function add_extra_headers(request) {
```

Comment:
Though you'll find much old JavaScript code here which uses the lowercase_and_underscores style for functions, variables, etc., new code should use headlessCamelCase:

http://guacamole.apache.org/guac-style/#naming

Please also be sure to document the parameters of the function, even for functions which are private.